src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -552,11 +557,12 @@ public void replaceVariablesInFileName(String startPath, String targetString, St
      * <p>
      * {@link #replaceVariablesInFile(String, Map) replaceVariablesInFile}
      *
-     * @param startPath    the path where the start directory is located
-     * @param replacements the replacements that should be applied
+     * @param startPath     the path where the start directory is located
+     * @param replacements  the replacements that should be applied
+     * @param filesToIgnore the name of files for which no replacement should be done
      * @throws IOException if an issue occurs on file access for the replacement of the variables.
      */
-    public void replaceVariablesInFileRecursive(String startPath, Map<String, String> replacements) throws IOException {
+    public void replaceVariablesInFileRecursive(String startPath, Map<String, String> replacements, List<String> filesToIgnore) throws IOException {
```

Comment:
I agree that this would be better style. But because of the recursive way the method is implemented, a lot of copy-pasting would be required that would outweigh the benefit of the overloading. What are your thoughts on that?

Reply:
Couldn’t the overloaded method just be defined as
```java
public void replaceVariablesInFileRecursive(String startPath, Map<String, String> replacements) throws IOException {
    replaceVariablesInFileRecursive(startPath, replacements, List.of());
}
```
Then no copy-pasting is required.

Reply:
Good idea, implemented. 👍